tls/s2n_security_policies.h

```diff
@@ -70,6 +70,11 @@ struct s2n_security_policy {
      * https://www.rfc-editor.org/rfc/rfc8446#section-4.2.7
      */
     const struct s2n_ecc_preferences *ecc_preferences;
+    /* This field determines what public keys are allowed for use. It restricts
+     * both the type of the key (Elliptic Curve, RSA w/ Encryption, RSA PSS) and
+     * the size of the key.
+     */
+    const struct s2n_certificate_key_preferences *certificate_key_preferences;
```

Comment:
Can you put a note on this field with a link to your issue for cleaning up the certificate_signature_preferences / s2n_certificate_key_preferences complexity?